schematic_api/api/routes.py

```diff
@@ -608,8 +620,48 @@ def get_viz_tangled_tree_layers(schema_url, figure_type):
 
     return layers[0]
 
+def download_manifest(input_token, manifest_id, new_manifest_name='', as_json=None):
+    """
+    Download a manifest based on a given manifest id. 
+    Args:
+        input_token: token of asset store
+        manifest_syn_id: syn id of a manifest
+        newManifestName: new name of a manifest that gets downloaded.
+        as_json: boolean; If true, return a manifest as a json. Default to True
+    Return: 
+        file path of the downloaded manifest
+    """
+    # call config_handler()
+    config_handler()
+
+    # use Synapse Storage
+    store = SynapseStorage(input_token=input_token)
+
+    # default as_json to True
+    if as_json is None:
+        as_json=True
+
+    # try logging in to asset store
+    try:
+        syn = store.login(input_token=input_token)
+    except Exception: 
+        raise SynapseAuthenticationError
```

Comment:
What is your rationale for this `try... except`? 

I think a `SynapseAuthenticationError` is very specific to not being able to log in (maybe due to wrong password).  There is also `SynapseNoCredentialsError` or `SynapseTimeoutError`, or `SynapseHTTPError`

Reply:
Thanks for the suggestion Tom. I think maybe `SynapseNoCredentialsError` is not needed for now since the token is a required parameter. But I could add other errors. If I have SynapseHTTPError, can I raise it as: 
```
    except SynapseHTTPError as e:
        raise(f"Please see Synapse HTTP error status:{e.response.status_code}")
```